### PR TITLE
Add e2e test for Lease API

### PR DESCRIPTION
**Pull request by wojtek-t**

### CODE REVIEW

File: test/e2e/common/lease.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+			LeaseTransitions:     pointer.Int32Ptr(1),
+		}
+
+		_, err = leaseClient.Update(createdLease)
+		framework.ExpectNoError(err, "updating Lease failed")
+
+		readLease, err = leaseClient.Get(name, metav1.GetOptions{})
+		framework.ExpectNoError(err, "couldn't read Lease")
+		framework.ExpectEqual(apiequality.Semantic.DeepEqual(createdLease.Spec, readLease.Spec), true)
+
```
Comment (liggitt) [inline test/e2e/common/lease.go]: also exercise patch (and ensure it was effective), list, and deletecollection (can create a second one with a specific label, then filter the list/deletecollection calls to that label to scope impact)

**Comment by liggitt** [comment]
additions look good, just noted a few other endpoints that need exercising

File: test/e2e/common/lease.go (hunk 1 above)
Comment (wojtek-t) [reply test/e2e/common/lease.go]: My feeling is that given that we don't have any logic different for this resource than for other resources we don't need to test everything.
But sure - I can add - will do it now.

**Comment by wojtek-t** [comment]
@liggitt - PTAL - I extended the test to exercise requested api calls.

**Pull request merged**